### PR TITLE
Align with new HLD schema

### PR DESCRIPTION
**Pull request by theasianpianist**
Align DASH API with newest HLD changes: https://github.com/sonic-net/DASH/pull/520 and https://github.com/sonic-net/SONiC/pull/1658

- Route group API changes
- Metering implementation changes
- PL/PL-NSG
- Deprecate/rename some fields for clarity

### CODE REVIEW

**Review by prsunny — APPROVED** [review]

File: proto/route.proto [hunk 1]
```diff
@@ -43,7 +43,7 @@ message Route {
 
 // ENI route table with CA prefix for packet Outbound
 message RouteKey {
-    string eni = 1;
+    string group_id = 1;
```
Comment (Pterosaur) [inline proto/route.proto:46]: Will this change break swss Azp?

Comment (theasianpianist) [reply proto/route.proto:46]: oh yeah good point, will put together a PR to update orchagent/swss to reflect the name change. Will hold off on merging this PR until the SWSS PR is ready.

**Review by Pterosaur — APPROVED** [review]

File: proto/eni.proto [hunk 2]
```diff
@@ -22,17 +22,14 @@ message Eni {
     // Vnet that ENI belongs to
     string vnet = 6;
     // Optional
-    // Private Link encoding for IPv6 SIP transpositions
-    optional types.IpPrefix pl_sip_encoding = 7;
```
Comment (prsunny) [inline proto/eni.proto:26]: Lets keep it here. Some bits seems to be required for datapath. We can update HLD later. @r12f for viz

Comment (r12f) [reply proto/eni.proto:26]: yea, in the latest design the pl_sip_encoding needs to be kept.

also changing the id for existing fields might not be a good idea, as it breaks ABI : D

File: proto/route.proto [hunk 3]
```diff
@@ -6,44 +6,42 @@ import "types.proto";
...
         string appliance = 4;
-        // service tunnel if action_type is {service_tunnel}
+        // service tunnel if routing_type is {service_tunnel}
         route.ServiceTunnel service_tunnel = 5;
     }
-    // Metering policy lookup enable (optional), default = false
-    optional bool metering_policy_en = 6;
-    // Metering class-id, used if metering policy lookup is not enabled
-    optional uint64 metering_class = 7;
+    optional uint32 metering_class_or = 6;
```
Comment (r12f) [inline proto/route.proto:38]: will it be better to use 8 here to avoid metering_policy_en and metering_class 7?

File: proto/route_rule.proto [hunk 4]
```diff
@@ -19,11 +19,12 @@ message RouteRule {
...
     // Optional
-    // Metering class-id
-    optional uint64 metering_class = 6;
+    // Metering class aggregate bits
+    optional uint32 metering_class_or = 6;
+    optional uint32 metering_class_and = 7;
     // Optional
     // optional region_id which the vni/prefix belongs to as a string for any vendor optimizations
-    optional string region = 7;
+    optional string region = 8;
```
Comment (r12f) [inline proto/route_rule.proto:27]: is this field needed? looks like we don't have this field in SAI.

File: proto/vnet_mapping.proto [hunk 5]
```diff
@@ -7,32 +7,28 @@ import "route_type.proto";
...
-    // override the metering class-id coming from the route table
-    optional bool override_meter = 5;
+    optional uint32 metering_class_or = 4;
     // Optional
     // if true, use the destination VNET VNI for encap. If false or not specified, use source VNET's VNI
-    optional bool use_dst_vni = 6;
+    optional bool use_dst_vni = 5;
     // Optional
-    optional bool use_pl_sip_eni = 7;
+    optional bool use_pl_sip_eni = 6;
```
Comment (r12f) [inline proto/vnet_mapping.proto:23]: do we need to remove this field?

File: proto/route.proto (hunk 1 above)
Comment (theasianpianist) [reply proto/route.proto]: deprecated ENI field and added group_id as a new field to avoid breaking change

File: proto/eni.proto (hunk 2 above)
Comment (theasianpianist) [reply proto/eni.proto:26]: deprecated this field instead of deleted

File: proto/route.proto (hunk 3 above)
Comment (theasianpianist) [reply proto/route.proto]: fixed, deprecated old metering fields

File: proto/route_rule.proto (hunk 4 above)
Comment (theasianpianist) [reply proto/route_rule.proto]: This field is present in the HLD, should we remove it from there if it's unneeded?

**Review by prabhataravind — APPROVED** [review]
lgtm

**Pull request merged**